src/internal_events/exec.rs

```diff
@@ -141,3 +141,80 @@ impl InternalEvent for ExecCommandExecuted<'_> {
         );
     }
 }
+
+pub enum ExecFailedToSignalChild {
+    #[cfg(unix)]
+    SignalError(nix::errno::Errno),
+    #[cfg(unix)]
+    FailedToMarshalPid(std::num::TryFromIntError),
+    #[cfg(unix)]
+    NoPid,
+    #[cfg(windows)]
+    IoError(std::io::Error),
+}
+
+impl ExecFailedToSignalChild {
+    fn to_error_code(&self) -> String {
```

Comment:
Is this not doing the same as `#[derive(Snafu)]` would provide with appropriate annotations?

Reply:
I started down this route but it actually felt messier to me since it required exporting each of the enum variants as a new type to be used in `sources::exec`. Unless you feel strongly, my preference would be to leave it as-is. I might be missing something though.

Also, just to clarify, Snafu would only help with the `Display` trait implementation here, right? Or would it also help with defining this additional `to_error_code` method?

Reply:
Ah, no, I misinterpreted what `to_error_code` was doing.